public/app/features/canvas/runtime/element.tsx

```diff
@@ -185,7 +185,7 @@ export class ElementState implements LayerElement {
     // Apply rotation (positive clockwise)
     style.transform = `translate(${translate[0]}, ${translate[1]}) rotate(${placement.rotation ?? 0}deg)`;
     // TODO determine rotation origin
-    //style.transformOrigin = `bottom left`;
+    style.transformOrigin = 'bottom left';
```

Comment:
Do you want bottom left for all element types?

Reply:
For now it works fine. I've got code floating around locally to make this an option in `CanvasElementOptions` because the ellipse/circle definitely won't be bottom left. We also probably don't want to set this for elements added from the UI (but if we had time making it configurable would be nifty).

Reply:
We decided to keep center transform origin and use the line midpoint to calculate the anchor point of the resultant element.